asn1ate/pyasn1gen.py

```diff
@@ -129,6 +129,10 @@ def __init__(self, sema_module, out_stream, referenced_modules):
     def generate_code(self):
         self.writer.write_line('# %s' % self.sema_module.name)
         self.writer.write_line('from pyasn1.type import univ, char, namedtype, namedval, tag, constraint, useful')
+
+        for imp, values in self.sema_module.imports.imports.iteritems():
+            self.writer.write_line('from _%s import %s' %
+                    ('_'.join([str(d.number) for d in imp.descendants() if isinstance(d, NameAndNumberForm)]),', '.join(map(_sanitize_identifier, values))))
         for module in self.referenced_modules:
```

Comment:
need to make some experiments

Reply:
I haven't found a quick way to solve this, help needed here

Reply:
I looked through this, and it seems like `referenced_modules` is the "module corpus" being defined by the ASN.1 input (i.e. there can be more than one module definition in the same input file).

So prior to your changes, we would only allow imports and type resolution between fully defined modules.

The pyasn1 generation would also generate a distinct Python module for each ASN.1 module. Since the ASN.1 was free to refer to types and values using e.g. `Module2.MyType` from `Module1.MyConstructedType`, the generated Python would add imports of all modules in the "corpus", so they were available.

Now you're trying to introduce something similar for `IMPORTS`. I think the difference is that there's not necessarily any ASN.1 definition available to asn1ate for an imported construct (there would need to exist some way to map an ASN.1 module ref to an external ASN.1 file, which would also need to be parsed, etc. None of this currently exists).

So I'm a little curious how well this `use_oid` flag generalizes. What happens for the existing tests in `testdata/imports.asn`? My guess is we will codegen imports of OID-named Python modules that don't exist (e.g. `_2_5_1_0_4.py` for `ImportMultipleWithAssignedOID` in that test file).

And not all `IMPORTS` are `NameAndNumberForm`, so the module names will not necessarily be complete/correct.

Hmm. This is all pretty complex, which is probably why I never got around to implementing it properly.

Reply:
It seems to me your changes must work for some particular case -- do you have a (possibly anonymized) ASN.1 input you could share that requires these changes and for which the generated pyasn1 is functional?

Reply:
my use case is to allow building a module without this having all the declaration available in the same file; now I see I misused the tool and this creates more problems than it solves (for my lack of knowledge). That said I think it's better to close this PR, unless you see anything useful.

Reply:
I'm thinking something must have gone right :). I like the use case of half-translating ASN.1 and leave some manual fixup to the user. It's much easier than doing it all by hand. But we want to make sure we don't silently generate something that's broken or make asn1ate more self-inconsistent.